packages/loader/container-loader/src/deltaQueue.ts

```diff
@@ -44,6 +44,12 @@ export class DeltaQueue<T> extends TypedEventEmitter<IDeltaQueueEvents<T>> imple
         return this.processingDeferred === undefined && this.q.length === 0;
     }
 
+    public async waitTillProcessingDone(): Promise<void> {
```

Comment:
rather than extending this interface can't we just may this a function
```waitTillProcessingDone(q){
  if(!q.idle){
    await new Promise((resolve,reject)=>{
      q.once("idle",resolve);
      q.once("error",reject);
    });
  }
}
```

Reply:
i prefer to keep the interface minimal unless changes are necessary

Reply:
I agree with you, but proposal has different semantics
That's because "idle" means empty queue. Someone can pause processing at any point in time and queue would not be idle.
we could add some other triggers, or adjust what "idle" actually means...

Reply:
"idle" / .idle is used. So we can add "working" / .working 
Or switch to using idle signal, though I'm afraid of someone pausing op processing and causing delay / deadlocks